### PR TITLE
Ceph 13 2 0 base

### PR DESCRIPTION
**Pull request by shangdehao1**
hi, @dtroyer @saulwold @mpeters-wrs @ian-jolliffewr 
     my teammate rebase some patches from stx-ceph10.2 patch list. 

please review it. 

thanks.

### CODE REVIEW

**Comment by iotcg** [comment]
These patches are from Wind River which were merged in Intel internal code base of 10.2.6 Ceph for startlingx. We're porting these patches for stx-ceph upgrade from 10.2.6 to 13.2.0. 
Please help review the patches.
1. Add .gitreview
2. Ceph Rebase: Changes for spec-include-TiS-changes.patch
3. Ceph Rebase: Changes for spec-disable-debuginfo-package.patch
4. Ceph Rebase: Changes for spec-add-init-script-patch.patch
5. Ceph Rebase: Changes for ceph-spec-add-service-files.patch
6. Ceph Rebase: Changes for tis-exclude-95-ceph-osd.rules-from-package.patch
7. Ceph Rebase: Changes for spec-Security-Gaps-File-permissions.patch
8. Ceph Rebase: Changes for fix-ceph-mon-started-twice.patch
9. Ceph Rebase: Changes for flag-ceph.conf-as-a-config-file.patch
10. Ceph Rebase: Changed for add-monitoring-and-recovery-of-hung-processe.patch
11. Ceph Rebase: Disable ceph user/group for Hammer equivalence.patch
12. Ceph Rebase: Add verbose rados and rbd exceptions.patch

There're some left patches that we're working on.

File: make-debs.sh [hunk 1]
```diff
@@ -33,7 +33,7 @@ git clean -dxf
 # c) compares higher than any previous commit
 # d) contains the short hash of the commit
 #
-vers=$(git describe --match "v*" | sed s/^v//)
+vers=`echo 'stx-ceph_v13.2.0' | sed 's/^stx-ceph_v//'`
```
Comment (dtroyer) [inline make-debs.sh]: Why bother with sed here if you're going to hard-code the response?

File: src/init-ceph.in [hunk 2]
```diff
@@ -447,6 +447,10 @@ for name in $what; do
 	    [ -n "$asok" ] && rm -f $asok
 	    [ -n "$post_stop" ] && do_cmd "$post_stop"
 	    [ -n "$lockfile" ] && [ "$?" -eq 0 ] && rm -f $lockfile
+            # flush journal to data disk in background
+            if [ "$type" == "osd" ];then
+            $(/usr/bin/ceph-osd -i $id --flush-journal) &
+            fi
```
Comment (dtroyer) [inline src/init-ceph.in]: is the indentation off here?

File: wrs/ceph.service [hunk 3]
```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=TiS Ceph Startup
```
Comment (dtroyer) [inline wrs/ceph.service]: Remove TiS

File: make-debs.sh (hunk 1 above)
Comment (iotcg) [reply make-debs.sh]: @dtroyer : 
1) Currently, it's hard-code here.
2) It's better to use tag to get this info
    git describe --match "v*" | sed 's/^v//'
3) There's no tag in stx-ceph v13.2.0 branch.
I think we could give a tag later.

File: wrs/ceph-radosgw.service [hunk 4]
```diff
@@ -0,0 +1,17 @@
+[Unit]
```
Comment (dtroyer) [inline wrs/ceph-radosgw.service]: Now seems like a good time to change this directory name to something that isn't wrs

File: make-debs.sh (hunk 1 above)
Comment (iotcg) [reply make-debs.sh]: @dtroyer : Do you agree with that we mark the tag as "stx-ceph_v13.2.0" in stx-ceph code base?

File: src/init-ceph.in [hunk 5]
```diff
@@ -451,6 +791,10 @@ for name in $what; do
             if [ "$type" == "osd" ];then
             $(/usr/bin/ceph-osd -i $id --flush-journal) &
             fi
+
+            wlog $name "INFO" "Process stopped, setting state to $ST_STOPPED"
+            save_proc_state $name $ST_STOPPED
```
Comment (dtroyer) [inline src/init-ceph.in]: Check indentation here too

**Comment by iotcg** [comment]
@dtroyer 
Please help review the patches.
1) code "indentation off" has been solved.

2) For the hard code "vers=`echo 'stx-ceph_v13.2.0' | sed 's/^stx-ceph_v//'`", I plan to solve it after all patches have been ported. The code repository need set a tag for stx-ceph_v13.2.0. Then we could use "git describe --match "v*" | sed 's/^v//'" to solve it.

3) "wrs" rename: good suggestion. Could we solve it later. Currently, "wrs" directory name is used in some ceph.spec or other files.

**Pull request closed without merging**